core/modules/layout/includes/block.text.inc

```diff
@@ -91,6 +91,18 @@ class BlockText extends Block {
           ),
         ),
       );
+      $form['convert']['description'] = array(
+       '#type' => 'textfield',
+       '#title' => t('Block description'),
+       '#maxlength' => 128,
+       '#description' => t('Used in administrative interfaces and not shown to end-users.'),
+       '#weight' => 3,
```

Comment:
This field needs to have a weight of 4. Otherwise, you end up in situations like this where the machine name comes after it when it should come after the admin label field:

![image](https://user-images.githubusercontent.com/2423362/27021313-cb2765a8-4f8a-11e7-8690-77fbe32911b5.png)